From 6dccc0dccb17c65d917d8c149335ea7f8983e22f Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 12 Oct 2015 04:46:22 -0400
Subject: [PATCH] src/cli/cli.c:fix compile failure

Tweak comparison to fix the failure:
...
error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
      || ((op != OPT_list_events) && (!(opts & OPTMASK_need_arg) == argc))
...

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/cli/cli.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cli/cli.c b/src/cli/cli.c
index 0072c10..0d396f6 100644
--- a/src/cli/cli.c
+++ b/src/cli/cli.c
@@ -105,7 +105,7 @@ int main(int argc, char** argv)
          * FALSE == FALSE (need arg but havent).
          * OPT_list_events is an exception, it can be used in both cases.
          */
-     || ((op != OPT_list_events) && (!(opts & OPTMASK_need_arg) == argc))
+     || ((op != OPT_list_events) && (argc == !(opts & OPTMASK_need_arg)))
     ) {
         show_usage_and_die(program_usage_string, program_options);
     }
-- 
1.9.1

